server/base/src/main/java/org/apache/accumulo/server/ServerInfo.java

```diff
@@ -57,18 +58,17 @@ public class ServerInfo implements ClientInfo {
   // set things up using the config file, the instanceId from HDFS, and ZK for the instanceName
   static ServerInfo fromServerConfig(SiteConfiguration siteConfig) {
     final Function<ServerInfo,String> instanceNameFromZk = si -> {
+      InstanceId iid = VolumeManager.getInstanceIDFromHdfs(
+          si.getServerDirs().getInstanceIdLocation(si.getVolumeManager().getFirst()),
+          si.getHadoopConf());
```

Comment:
By merging this instanceId lookup into the function to look up the instanceName, and removing the function to track the instanceId, that creates an unnecessary need to look up the instanceId in ZooKeeper later.

One of the goals of the previous iteration I did in here was to make sure that we don't do unnecessary lookups like that. The ServerInfo should already have the instanceId from the volume manager lookup, so when ServerContext asks ServerInfo for it, it should never need to go back to ZooKeeper, since it already has it. Here, it goes to the volume manager, but then doesn't remember it, so ZooKeeper must be consulted later.